### PR TITLE
Clustering and SBR documentation

### PR DESCRIPTION
**Pull request by fsat**
Additional entries in the documentation which talks about:
- Considerations in planning for a resilient cluster setup
- Recovery steps in the event of cluster split
- Recovery steps in the event of whole cluster down

### CODE REVIEW

**Comment by fsat** [comment]
**Important:** Merge in conjunction with the release of https://github.com/typesafehub/conductr/pull/700

File: src/main/play-doc/intro/Clustering-Considerations.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Typesafe ConductR %PLAY_VERSION%
+
+## Cluster Resiliency Considerations
+
+In order to achieve cluster resilience when using ConductR, it’s recommended for the cluster to be separated into at least 3 distinct parts, e.g.
+
```
Comment (fsat) [inline src/main/play-doc/intro/Clustering-Considerations.md]: I can't think of better words than "3 distinct parts" - better suggestions are appreciated

File: src/main/play-doc/intro/Clustering-Considerations.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+The `keep-majority` strategy ensures when cluster split occurs:
+
+* The partition which has the majority of the nodes are kept running
+* Other partition(s) which has less than majority of the nodes will have each of their nodes terminated gracefully.
+
+## Cluster Split Automated Recovery
+
+In the RPM version of the ConductR, `systemd` is utilised to ensure the ConductR process keeps running. Akka SBR and `systemd` in conjunction will allow ConductR to perform automated recovery should the event of cluster split occur.
+
```
Comment (fsat) [inline src/main/play-doc/intro/Clustering-Considerations.md]: Making sure 'Restart' is configured in ConductR RPM's systemd is out of the outstanding task still needs to be done

**Comment by fsat** [comment]
@huntc - can I get your review on these please?

**Comment by fsat** [comment]
@edwardcallahan - since you're working on the RPM, would you be able to give this document a quick glance please? Particularly section about "Cluster Split Automated Recovery" where there's mention about systemd

**Comment by fsat** [comment]
@huntc, @markusjura, and @edwardcallahan - can I get your feedback on the docs please?

File: src/main/play-doc/operation/ClusterTroubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Cluster Troubleshooting
+
+Cluster partition is a failure that may occur when running a cluster based application. To recover from this situation ConductR utilizes Akka SBR (Split Brain Recovery). Akka SBR is a part of [Akka Typesafe Reactive Platform](http://www.typesafe.com/products/typesafe-reactive-platform) version `15v01p05`.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/ClusterTroubleshooting.md]: It should be Typesafe Reactive Platform not Akka Typesafe Reactive Platform. We could say part of Akka in Typesafe Reactive Platform version or the likes, but the platform is TRP.

File: src/main/play-doc/operation/ClusterTroubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+* If the service has been restarted successfully continue with the next node and repeat that until all services has been restarted.
+
+**Notes on `seed-nodes` file**
+* The address of last known members in the cluster are kept in what we call seed nodes file.
+* The file is located at `/usr/share/conductr/conf/seed-nodes`.
+* By default ConductR keeps the latest 3 reachable members in the cluster in the seed nodes file.
+
+## Whole Cluster Termination Recovery
+
+Should the cluster split occurs in such way that each partition does not have majority of the node, all ConductR process within the cluster will be terminated.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/ClusterTroubleshooting.md]: Maybe link to ClusterSetupConsiderations i.e. Deploying a cluster across at least 3 network partitions improves the chances of maintaining a majority and preventing cluster shutdown. See ClusterSetupConsiderations.md for more information.

also "split occurs" -> "split  occur"

File: src/main/play-doc/operation/ClusterTroubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+An example of this would be a cluster with 3 nodes, and the split occurs in such a way that none of the nodes are able to see each other. In this situation, each partition will have 1 node each, causing all ConductR process to be terminated.
+
+The recovery in this situation can be achieved by restarting the whole ConductR cluster.
+
+## Restarting the ConductR Cluster
+
+Restarting the entire ConductR cluster should only be considered in a face of disaster where all nodes have been lost. This restart need to be performed in a sequential order.
+
+**Steps**
+* Restart the `conductr` service first on one node.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/ClusterTroubleshooting.md]: Remove the seed-nodes file and restart the `conductr` service on one node.

File: src/main/play-doc/operation/ClusterTroubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+```bash
+sudo service conductr stop
+sudo rm /usr/share/conductr/conf/seed-nodes
+sudo service conductr start
+```
+* *Ensure the ConductR process has been started successfully on this node.* Upon restart, the node will not find any existing seed node due to the fact the `seed-nodes` file has been removed. Therefore the node will register itself as a seed node in a new cluster.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/ClusterTroubleshooting.md]: How do I ensure successful start?

File: src/main/play-doc/operation/ClusterSetupConsiderations.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Cluster Resiliency Considerations
+
+In order to achieve cluster resilience, it’s recommended for the cluster to be separated into at least 3 distinct parts, e.g.
+
+* If you are using Amazon, this would be spreading the number of nodes evenly in 3 different availability zones.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/ClusterSetupConsiderations.md]: Amzon -> Amazon Web Services or AWS EC2

**Comment by fsat** [comment]
The SBR feature has been released as part of ConductR 1.0.6 - this document can be merged back into master.

**Pull request merged**